### ISSUE TITLE
ToDO: missing red text on NULL fields in 'Add Tree' AND enable user to save data entries and enter another

### ISSUE THREAD

**Issue by TreeMama** [TreeMama/iSeaTree-React-Prototype#391]
ToDO : Our current build has 1 shipblocker.  It can be described in two parts.

1) For some reason there is missing red text for the NULL fields - note: this was entered FOR EVERY FIELD and has been working since the completion on #34 .)

![86881041-777ae880-c0a2-11ea-9656-7609d7642ece](https://user-images.githubusercontent.com/62637055/213297803-904f7701-b48c-497c-9f39-ac873090600d.png)

2) Also odd - this was captured by the same Android user - using the below dropbox video.... but I'll explain here as well.

If you fill out the form to add a tree, and fill it out completely, you can Calculate Tree Benefits and Save with no issue as many times as you'd like. 

**However**, _if you forget to fill out one of the fields and click Calculate Tree Benefits and Save, nothing happens and no error is visible_(see part 1 of this ticket). 

From that point on, the Calculate Tree Benefits and Save button does not appear to do anything,  even if you fill out the form completely. Even if you clear out the current form and start over, it still won't do anything. The only way to get it working again is to restart the app. 

Here is a video (with audio) of the error:

https://www.dropbox.com/s/1w8ebfwy71ry1aq/IMG_8189.MOV?dl=0

**Comment by rimzici** [comment]
Hi @bestw07  @TreeMama 
Have a better viable fix. Please accept my(Rimnesh Fernandez) Upwork proposal and I will be happy to share.
Thanks

**Comment by rimzici** [comment]
Pardon me. I mean no offence to any.

![Screenshot 2023-01-19 at 9 50 59 AM](https://user-images.githubusercontent.com/16771765/213488650-13d003be-ac2f-4d7a-89d6-9a4fae0114bb.png)

The change made above will fix the 2nd issue - Could add a tree first time, but if tried to add again after that without killing the app, nothing happens.

Justification: It is because 
```javascript
useEffect(() => {
    ; (async function () {
      console.log('loadBenefitsCall ===', loadBenefitsCall)
      if (loadBenefitsCall) {
        await loadBenefits()
      }
    })()
  }, [loadBenefitsCall])
```
This method will only be triggered if `loadBenefitsCall` value has changed.
On initial render the `loadBenefitsCall` is `false`. When first tree is added it is changed to `true`, but when tried to add a tree again, it is again set to `true`. true -> true - no change. Hence `await loadBenefits()` will not be called.

**Comment by TreeMama** [comment]
appears resolved as of 1/19/23 v 4.0.2 (69)
